File: web/src/app/Models/Dtos/Test/question-result-dto.ts
import { QuestionType } from '../../Enums/QuestionType';

export interface QuestionResultDto {
  questionId: string;
  questionTitle: string;
  type: QuestionType;
  isCorrect: boolean;
  pointsAwarded: number;
  maxPoints: number;
  userResponse: string;
  aiFeedback: string;
}

File: backend/SkillProof/SkillProof.Entities/Dtos/Tests/QuestionResultDto.cs
using SkillProof.Entities.Enums;

namespace SkillProof.Entities.Dtos.Tests;

public class QuestionResultDto
{
    public string QuestionId { get; set; } = string.Empty;
    public string QuestionTitle { get; set; } = string.Empty;
    public QuestionType Type { get; set; }
    public bool IsCorrect { get; set; }
    public int PointsAwarded { get; set; }
    public int MaxPoints { get; set; }
    public string UserResponse { get; set; } = string.Empty;
    public string AiFeedback { get; set; } = string.Empty;
}
